src/function/finitediff.jl

```diff
@@ -56,7 +56,7 @@ function instantiate_function(f, x, adtype::AutoFiniteDiff, p, num_cons=0)
     end
 
     if f.hess === nothing
-        hess = (res, θ, args...) -> FiniteDiff.finite_difference_hessian!(res, x -> _f(x, args...), θ, FiniteDiff.HessianCache(x, adtype.fdhtype))
+        hess = (res, θ, args...) -> FiniteDiff.finite_difference_hessian!(res, x -> _f(x, args...), θ, FiniteDiff.HessianCache(θ, adtype.fdhtype))
```

Comment:
We should also cache the caches.

Reply:
Do you mean that instead of creating a new one for every call reset it and use here and have it defined once only?